protocol/auditor.go

```diff
@@ -0,0 +1,176 @@
+// This module implements a generic CONIKS auditor, i.e. the
+// functionality that clients and auditors need to verify
+// a server's STR history.
+
+package protocol
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+)
+
+// Auditor provides a generic interface allowing different
+// auditor types to implement specific auditing functionality.
+type Auditor interface {
```

Comment:
Tbh, the way it's implemented right now was according to a suggestion you made to make `AudState` implement `Auditor` and have the other two use `AudState` so that other kinds of auditors could implement something different. I'm a bit confused why you're suggesting now that we should change this again ;)

I need to think about it some more, but I think that the only new type of auditor which would have to implement the `Auditor` interface would be a blockchain based auditor. So I don't think we need to change what we have when it already expresses what we intended. Maybe @arlolra and @Liamsi can weigh in on this, too?